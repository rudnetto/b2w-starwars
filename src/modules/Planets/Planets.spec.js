import React from 'react';
import { shallow } from 'enzyme';

const planetMock = {
  name: 'Tatooine',
  population: '120000',
  terrain: 'Dessert',
  climate: 'Arid',
  filmsFeatured: 5,
};

describe('Planets test suite', () => {
  it('should render', () => {
    jest.doMock('./Service', () => ({ getRandomPlanet: jest.fn(() => Promise.resolve(planetMock)) }));

    const Planets = require('./Planets').default; // eslint-disable-line
    const element = shallow(<Planets />);
    expect(element).toMatchSnapshot();
  });
});
