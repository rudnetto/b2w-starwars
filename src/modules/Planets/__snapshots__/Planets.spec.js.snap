// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Planets test suite should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Planets />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <styled.div>
        <styled.div>
                <Styled(WithStyles(Icon))
                        color="primary"
                        style={
                                Object {
                                        "fontSize": 64,
                                      }
                        }
                >
                        loop
                </Styled(WithStyles(Icon))>
        </styled.div>
</styled.div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <styled.div>
          <Styled(WithStyles(Icon))
                    color="primary"
                    style={
                              Object {
                                        "fontSize": 64,
                                      }
                    }
          >
                    loop
          </Styled(WithStyles(Icon))>
</styled.div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            false,
            <Styled(WithStyles(Icon))
              color="primary"
              style={
                            Object {
                                          "fontSize": 64,
                                        }
              }
>
              loop
</Styled(WithStyles(Icon))>,
            false,
          ],
        },
        "ref": null,
        "rendered": Array [
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "loop",
              "color": "primary",
              "style": Object {
                "fontSize": 64,
              },
            },
            "ref": null,
            "rendered": "loop",
            "type": [Function],
          },
          false,
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <styled.div>
          <styled.div>
                    <Styled(WithStyles(Icon))
                              color="primary"
                              style={
                                        Object {
                                                  "fontSize": 64,
                                                }
                              }
                    >
                              loop
                    </Styled(WithStyles(Icon))>
          </styled.div>
</styled.div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <styled.div>
            <Styled(WithStyles(Icon))
                        color="primary"
                        style={
                                    Object {
                                                "fontSize": 64,
                                              }
                        }
            >
                        loop
            </Styled(WithStyles(Icon))>
</styled.div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              false,
              <Styled(WithStyles(Icon))
                color="primary"
                style={
                                Object {
                                                "fontSize": 64,
                                              }
                }
>
                loop
</Styled(WithStyles(Icon))>,
              false,
            ],
          },
          "ref": null,
          "rendered": Array [
            false,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "loop",
                "color": "primary",
                "style": Object {
                  "fontSize": 64,
                },
              },
              "ref": null,
              "rendered": "loop",
              "type": [Function],
            },
            false,
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
